libclc/amdgcn-amdhsa/libspirv/atomic/atomic_add.cl

```diff
@@ -10,13 +10,41 @@
 #include <spirv/spirv.h>
 #include <spirv/spirv_types.h>
 
+extern constant int __oclc_ISA_version;
+
 AMDGPU_ATOMIC(_Z18__spirv_AtomicIAdd, int, i, __hip_atomic_fetch_add)
 AMDGPU_ATOMIC(_Z18__spirv_AtomicIAdd, unsigned int, j, __hip_atomic_fetch_add)
 AMDGPU_ATOMIC(_Z18__spirv_AtomicIAdd, long, l, __hip_atomic_fetch_add)
 AMDGPU_ATOMIC(_Z18__spirv_AtomicIAdd, unsigned long, m, __hip_atomic_fetch_add)
-AMDGPU_ATOMIC(_Z21__spirv_AtomicFAddEXT, float, f, __hip_atomic_fetch_add)
 
-#define AMDGPU_ATOMIC_FP64_ADD_IMPL(AS, AS_MANGLED, SUB1, SUB2)                                                                 \
+#define AMDGPU_ATOMIC_FP32_ADD_IMPL(AS, AS_MANGLED, SUB1, CHECK, NEW_BUILTIN)                              \
```

Comment:
So `CHECK` here is because we compile for `gfx940` but might be using it with lower ISA versions, and then fall back to different impl?

Reply:
We compile libclc for gfx940 in order to enable all of the gfx940 builtins. However, we do not get the definition of 
```
extern constant int __oclc_ISA_version;
```
Until we compile kernel code, and the bc file that contains this definition will be linked:
``
$ clang++ -fsycl -fsycl-targets=amd_gpu_gfx90a ~/test.cpp -###
"/home/hugh/llvm/build/bin/clang-16" "-cc1" ....
"-mlink-builtin-bitcode" "/opt/rocm/amdgcn/bitcode/oclc_isa_version_90a.bc"
```
Which allows the branch to be taken or not at compile time

Reply:
Cool, a nitpick I would rename it to something like `ARCH_CHECK` to make it obvious that it's not a correctness thing.

Reply:
Sure change made. Have added check macro funcs `AMDGPU_ARCH_BETWEEN` and `AMDGPU_ARCH_GEQ`